custom_components/tahoma/switch.py

```diff
@@ -31,47 +32,66 @@ class TahomaSwitch(TahomaDevice, SwitchEntity):
     def __init__(self, tahoma_device, controller):
         """Initialize the switch."""
         super().__init__(tahoma_device, controller)
-        self._state = STATE_OFF
-        self._skip_update = False
+
+        self._state = None
 
     def update(self):
         """Update method."""
-        # Postpone the immediate state check for changes that take time.
+
         if self.should_wait():
             self.schedule_update_ha_state(True)
             return
 
-        if self._skip_update:
-            self._skip_update = False
-            return
-
         self.controller.get_states([self.tahoma_device])
 
-        _LOGGER.debug("Update %s, state: %s", self._name, self._state)
+        if CORE_ON_OFF_STATE in self.tahoma_device.active_states:
+            self.current_value = (
+                self.tahoma_device.active_states.get(CORE_ON_OFF_STATE) == "on"
+            )
 
     @property
     def device_class(self):
         """Return the class of the device."""
 
+        if self.tahoma_device.uiclass == "Siren":
+            return DEVICE_CLASS_SIREN
+
         return DEVICE_CLASS_SWITCH
 
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+
+        if self.device_class == DEVICE_CLASS_SIREN:
+            if self.is_on:
+                return "mdi:bell-ring"
+            else:
+                return "mdi:bell-off"
+
+        return None
+
     def turn_on(self, **kwargs):
         """Send the on command."""
-        _LOGGER.debug("Turn on: %s", self._name)
 
-        self.apply_action("on")
-        self._skip_update = True
-        self._state = STATE_ON
+        if "on" in self.tahoma_device.command_definitions:
+            return self.apply_action("on")
+
+        if "ringWithSingleSimpleSequence" in self.tahoma_device.command_definitions:
+            return self.apply_action(
+                "ringWithSingleSimpleSequence", 120000, 75, 2, "memorizedVolume"
```

Comment:
These are adhoc values taken from the doc if I recall correctly. Do we agree to keep those until we understand better what it does?

Reply:
Is there a risk of us breaking devices? If not, we could maybe keep this in and listen to feedback at some point. Or we could disable the Siren as a switch in the const for now.

Reply:
I don't think it can break anything, aside from someone's ears...

Reply:
It would be really nice to have a comment to explain where these values come from.